components/kyma-environment-broker/internal/process/provisioning/enable_for_trial_step.go

```diff
@@ -0,0 +1,45 @@
+package provisioning
+
+import (
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/broker"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/process"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage"
+)
+
+type EnableForTrialPlanStep struct {
+	step             Step
+	operationManager *process.ProvisionOperationManager
+}
+
+// ensure the interface is implemented
+var _ Step = (*EnableForTrialPlanStep)(nil)
+
+func NewEnableForTrialPlanStep(os storage.Operations, step Step) *EnableForTrialPlanStep {
+	return &EnableForTrialPlanStep{
+		step:             step,
+		operationManager: process.NewProvisionOperationManager(os),
+	}
+}
+
+func (s *EnableForTrialPlanStep) Name() string {
+	return s.step.Name()
+}
+
+func (s *EnableForTrialPlanStep) Run(operation internal.ProvisioningOperation, log logrus.FieldLogger) (internal.ProvisioningOperation, time.Duration, error) {
+	pp, err := operation.GetProvisioningParameters()
```

Comment:
Once **EnableComponent** feature is available. we'll have to enable the disabled optional components. in this case NAts

Reply:
Yes, but the controlling step is done, like the other steps in "main.go", using EnableForTrialPlanStep as a generic step:
```
		{
			weight: 2,
			step: provisioning.NewEnableForTrialPlanStep(db.Operations(),
				provisioning.NewNatsStreamingStep(db.Operations())),
		},
		{
			weight: 2,
			step: provisioning.NewEnableForTrialPlanStep(db.Operations(),
				provisioning.NewKnativeProvisionerNatssStep(db.Operations())),
		},
```

Reply:
from what i understand, the controlling step merely executes the step. but enabling the component for installation needs to be done explicitly in the step using some parameter.

Reply:
Till now, they are. two generic "controlling" steps: "Skip" and the new one coming from us "Enable". See again in main:
```
		{
			weight: 2,
			step: provisioning.NewSkipForTrialPlanStep(db.Operations(),
				provisioning.NewProvisionAzureEventHubStep(db.Operations(), azure.NewAzureProvider(), accountProvider, ctx)),
		},
		{
			weight: 2,
			step: provisioning.NewEnableForTrialPlanStep(db.Operations(),
				provisioning.NewNatsStreamingStep(db.Operations())),
		},
```